internal/transport/http2_client.go

```diff
@@ -1438,7 +1441,11 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 	}
 
 	if statusGen == nil {
-		statusGen = status.New(rawStatusCode, grpcMessage)
+		if rawStatusCode == nil {
+			rsc := codes.Unknown
+			rawStatusCode = &rsc
+		}
+		statusGen = status.New(*rawStatusCode, grpcMessage)
```

Comment:
Let's do this the other way around to avoid an allocation (even if it hopefully doesn't escape the stack):

```go
if statusGen == nil {
	rsc := codes.Unknown
	if rawStatusCode != nil {
		rsc = *rawStatusCode
	}
	statusGen = status.New(rsc, grpcMessage)
}
```

Reply:
Done!